src/nbl/video/CVulkanCommandBuffer.h

```diff
@@ -1083,19 +1099,38 @@ class CVulkanCommandBuffer : public IGPUCommandBuffer
                 // Do a look ahead
                 if ((i + 1 > descriptorSetCount - 1) || !pDescriptorSets[i + 1])
                 {
-                    vk->vk.vkCmdBindDescriptorSets(
-                        m_cmdbuf,
-                        static_cast<VkPipelineBindPoint>(pipelineBindPoint),
-                        vk_pipelineLayout,
-                        firstSet+first, last - first, vk_descriptorSets+first, dynamicOffsetCount, dynamicOffsets);
+                    if (dynamicOffsets)
+                    {
+                        uint32_t totalDynamicOffsetCount = 0u;
+                        for (uint32_t setIndex = first; setIndex < last; ++setIndex)
+                            totalDynamicOffsetCount += dynamicOffsetCountPerSet[setIndex];
+
+                        vk->vk.vkCmdBindDescriptorSets(
+                            m_cmdbuf,
+                            static_cast<VkPipelineBindPoint>(pipelineBindPoint),
+                            vk_pipelineLayout,
+                            // firstSet + first, last - first, vk_descriptorSets + first, vk_dynamicOffsetCount, vk_dynamicOffsets);
+                            firstSet + first, last - first, vk_descriptorSets + first,
+                            totalDynamicOffsetCount, dynamicOffsets + dynamicOffsetsBindOffset);
+
+                        dynamicOffsetsBindOffset += totalDynamicOffsetCount;
+                    }
+                    else
+                    {
+                        vk->vk.vkCmdBindDescriptorSets(
+                            m_cmdbuf,
+                            static_cast<VkPipelineBindPoint>(pipelineBindPoint),
+                            vk_pipelineLayout,
+                            firstSet+first, last - first, vk_descriptorSets+first, 0u, nullptr);
+                    }
+
                     first = ~0u;
                     last = ~0u;
+                    ++bindCount;
                 }
             }
         }
-
-        // vk->vk.vkCmdBindDescriptorSets(m_cmdbuf, static_cast<VkPipelineBindPoint>(pipelineBindPoint),
-        //     vk_pipelineLayout, firstSet, descriptorSetCount, vk_descriptorSets, dynamicOffsetCount, dynamicOffsets);
+        assert(bindCount <= core::ceil(MAX_DESCRIPTOR_SET_COUNT / 2.f));
```

Comment:
@nipunG314 this function got completely rewritten since then, you can discard all changes to this file